clients/fides-js/src/lib/gpp/us-notices.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Helper functions to set the GPP CMP API based on Fides values
+ */
+
+import { CmpApi } from "@iabgpp/cmpapi";
+
+import { FidesCookie } from "../cookie";
+import {
+  FIDES_REGION_TO_GPP_SECTION,
+  NOTICE_KEY_TO_FIDES_REGION_GPP_FIELDS,
+} from "./constants";
+import { PrivacyExperience } from "../consent-types";
+
+/**
+ * Sets the appropriate fields on a GPP CMP API model for whether notices were provided
+ *
+ * Returns GPP section IDs which were updated
+ */
+export const setGppNoticesProvidedFromExperience = ({
+  cmpApi,
+  experience,
+}: {
+  cmpApi: CmpApi;
+  experience: PrivacyExperience;
+}) => {
+  const sectionsChanged = new Set<{ name: string; id: number }>();
+  const { privacy_notices: notices, region } = experience;
+
+  Object.entries(NOTICE_KEY_TO_FIDES_REGION_GPP_FIELDS).forEach(
+    ([noticeKey, regionGppFields]) => {
+      const gppFields = regionGppFields[region];
+      if (gppFields) {
+        const { gpp_notice_fields: fields } = gppFields;
+        const experienceHasNotice = !!notices?.find(
+          (n) => n.notice_key === noticeKey
+        );
+        // 1 = Yes notice was provided, 2 = No, notice was not provided.
+        const value = experienceHasNotice ? 1 : 2;
+        const gppSection = FIDES_REGION_TO_GPP_SECTION[region];
+        sectionsChanged.add(gppSection);
+        fields.forEach((field) => {
+          cmpApi.setFieldValue(gppSection.name, field, value);
+        });
+      }
+    }
+  );
+
+  return Array.from(sectionsChanged);
+};
+
+/**
+ * Sets the appropriate fields on a GPP CMP API model for user opt-outs
+ *
+ * Returns GPP sections which were updated
+ */
+export const setGppOptOutsFromCookie = ({
```

Comment:
we iterate through all possible notice types and set the GPP "mechanism" fields (i.e. `SaleOptOut`, _not_ `SaleOptOutNotice`) according to whether or not that notice has been opted into/out of on the cookie